rs/tests/networking/state_sync_performance.rs

```diff
@@ -0,0 +1,214 @@
+/* tag::catalog[]
+
+Title:: State Sync performance test
+
+Runbook::
+. setup a testnet with a single node
+. install nns and state sync canisters
+. grow the state of the subnet by expanding states of the state sync canisters
+. add 12 additional nodes
+. fetch state sync durations of joining nodes via metrics
+
+end::catalog[] */
+
+use anyhow::Result;
+use canister_test::Canister;
+use futures::future::join_all;
+use ic_consensus_threshold_sig_system_test_utils::execute_proposal;
+use ic_nns_constants::GOVERNANCE_CANISTER_ID;
+use ic_nns_governance_api::NnsFunction;
+use ic_registry_subnet_type::SubnetType;
+use ic_system_test_driver::{
+    driver::{
+        farm::HostFeature,
+        group::SystemTestGroup,
+        ic::{ImageSizeGiB, InternetComputer, Subnet, VmResources},
+        prometheus_vm::{HasPrometheus, PrometheusVm},
+        simulate_network::{FixedNetworkSimulation, SimulateNetwork},
+        test_env::TestEnv,
+        test_env_api::{
+            HasPublicApiUrl, HasTopologySnapshot, IcNodeContainer, NnsInstallationBuilder,
+        },
+    },
+    systest,
+    util::{block_on, runtime_from_url},
+};
+use ic_types::Height;
+use registry_canister::mutations::do_add_nodes_to_subnet::AddNodesToSubnetPayload;
+use rejoin_test_lib::{
+    assert_state_sync_has_happened, install_statesync_test_canisters, modify_canister_heap,
+};
+use slog::info;
+use std::time::Duration;
+
+const INITIAL_NODES: usize = 1;
+const TOTAL_NODES: usize = 13;
+
+const BANDWIDTH_MBITS: u32 = 300; // artificial cap on bandwidth
+const LATENCY: Duration = Duration::from_millis(150); // artificial added latency
+
+const SIZE_LEVEL: usize = 1;
+const NUM_CANISTERS: usize = 4;
+
+pub const SUCCESSFUL_STATE_SYNC_DURATION_SECONDS_SUM: &str =
+    "state_sync_duration_seconds_sum{status=\"ok\"}";
+pub const SUCCESSFUL_STATE_SYNC_DURATION_SECONDS_COUNT: &str =
+    "state_sync_duration_seconds_count{status=\"ok\"}";
+
+fn setup(env: TestEnv) {
+    PrometheusVm::default()
+        .start(&env)
+        .expect("failed to start prometheus VM");
+
+    InternetComputer::new()
+        .with_default_vm_resources(VmResources {
+            boot_image_minimal_size_gibibytes: Some(ImageSizeGiB::new(
+                30 + 2 * NUM_CANISTERS as u64 * SIZE_LEVEL as u64,
+            )),
+            ..VmResources::default()
+        })
+        .with_required_host_features(vec![HostFeature::IoPerformance])
```

Comment:
The 5 [io-performance hosts](https://github.com/dfinity-lab/farm/blob/master/dcs.tf#L183-L188) were originally intended for manual use by the messaging and runtime teams and not for running CI jobs on. I'm OK with using them for CI if those teams are OK with it as well. Note that every time the `Release Testing` workflow is run (either periodically or when manually triggered, like this morning) these io-performance host will be acquired.

Alternatively you could consider requiring `HostFeature::Performance` of which we have [way more hosts](https://github.com/dfinity-lab/farm/blob/master/dcs.tf#L193-L236).